Laboratory/TypeScript/test/flags.ts

```diff
@@ -0,0 +1,7 @@
+import * as G from './generated/gen';
+
+it("Supports flag enums", () => {
```

Comment:
Maybe it would be good to add a const "isFlagsEnum: bool" to the enum definitions

Something like

```ts
export enum MyEnum {
 // ...
}

export const $MyEnumIsFlags = true;
```

Or maybe injecting it into the enum object and updating the typing to reflect? This seems a little harder though.
```ts
enum _MyEnum {...}
(_MyEnum as any).isFlags = true;
type MyEnumType = typeof _MyEnum & { isFlags: true }
export const MyEnum = _MyEnum as unknown as MyEnumType
```

I think that would work?